plugin/pkg/scheduler/algorithm/priorities/interpod_affinity.go

```diff
@@ -98,99 +87,85 @@ func (ipa *InterPodAffinity) CalculateInterPodAffinityPriority(pod *api.Pod, nod
 	}
 
 	// convert the topology key based weights to the node name based weights
-	var maxCount int
-	var minCount int
-	counts := map[string]int{}
-	for _, node := range nodes {
-		totalCount := 0
-		// count weights for the weighted pod affinity
-		if affinity.PodAffinity != nil {
-			for _, weightedTerm := range affinity.PodAffinity.PreferredDuringSchedulingIgnoredDuringExecution {
-				weightedCount, err := ipa.CountWeightByPodMatchAffinityTerm(pod, allPods, weightedTerm.Weight, weightedTerm.PodAffinityTerm, node)
-				if err != nil {
-					return nil, err
+	var maxCount float64
+	var minCount float64
+	counts := make(map[string]float64, len(nodes))
+
+	processTerm := func(term *api.PodAffinityTerm, affinityPod, podToCheck *api.Pod, fixedNode *api.Node, weight float64) error {
+		match, err := podMatchesNamespaceAndSelector(podToCheck, affinityPod, term)
+		if err != nil {
+			return err
+		}
+		if match {
+			for _, node := range nodes {
+				if ipa.failureDomains.NodesHaveSameTopologyKey(node, fixedNode, term.TopologyKey) {
+					counts[node.Name] += weight
 				}
-				totalCount += weightedCount
 			}
 		}
-
-		// count weights for the weighted pod anti-affinity
-		if affinity.PodAntiAffinity != nil {
-			for _, weightedTerm := range affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution {
-				weightedCount, err := ipa.CountWeightByPodMatchAffinityTerm(pod, allPods, (0 - weightedTerm.Weight), weightedTerm.PodAffinityTerm, node)
-				if err != nil {
-					return nil, err
-				}
-				totalCount += weightedCount
+		return nil
+	}
+	processTerms := func(terms []api.WeightedPodAffinityTerm, affinityPod, podToCheck *api.Pod, fixedNode *api.Node, multiplier int) error {
+		for _, weightedTerm := range terms {
+			if err := processTerm(&weightedTerm.PodAffinityTerm, affinityPod, podToCheck, fixedNode, float64(weightedTerm.Weight*multiplier)); err != nil {
+				return err
 			}
 		}
+		return nil
+	}
 
-		// reverse direction checking: count weights for the inter-pod affinity/anti-affinity rules
-		// that are indicated by existing pods on the node.
-		for _, ep := range allPods {
-			epAffinity, err := api.GetAffinityFromPodAnnotations(ep.Annotations)
-			if err != nil {
+	for _, existingPod := range allPods {
+		existingPodNode, err := ipa.info.GetNodeInfo(existingPod.Spec.NodeName)
+		if err != nil {
+			return nil, err
+		}
+		existingPodAffinity, err := api.GetAffinityFromPodAnnotations(existingPod.Annotations)
+		if err != nil {
+			return nil, err
+		}
+
+		// Check whether <existingPod> matches <pod> affinity/antiaffinity terms.
+		if affinity.PodAffinity != nil {
+			terms := affinity.PodAffinity.PreferredDuringSchedulingIgnoredDuringExecution
```

Comment:
Here I'd make a more detailed comment, like
For every soft pod affinity term of \<pod\>, if \<existingPod\> matches the term, increment \<counts\> for every node in the cluster with the same term.TopologyKey value as that of \<existingPod\s>'s node by the term's weight.
(I know comments aren't supposed to say "how" but the code is sufficiently complicated that I think this merits an exception.)